CMakeLists.txt

```diff
@@ -24,3 +24,7 @@ target_link_libraries(ByteBufferTestExecutable SuperLib)
 add_library(SuperLib STATIC "src/SerializationHelper.cpp" "src/ByteBuffer.cpp")
 target_include_directories(SuperLib PUBLIC "include")
 set_target_properties(SuperLib PROPERTIES CXX_STANDARD 20)
+
+add_executable(MessageGenerator "src/GenerateMessageFile.cpp" "src/MessageCreator.cpp")
```

Comment:
To use a different c++ compiler the only thing that needs to change is how you first invoke CMake. Use the `CXX` environment variable to tell CMake where the compiler is.
```
CXX=/opt/homebrew/Cellar/gcc/13.2.0/bin/g++-13 cmake ..
```